Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/shape_smoothing.h

```diff
@@ -0,0 +1,413 @@
+
+#if defined(CGAL_EIGEN3_ENABLED)
+    CGAL_static_assertion_msg(
+      (!boost::is_same<typename GetSolver<NamedParameters, Default_solver>::type, bool>::value) || EIGEN_VERSION_AT_LEAST(3, 2, 0),
+      "Eigen3 version 3.2 or later is required.");
+#else
+    CGAL_static_assertion_msg(
+      (!boost::is_same<typename GetSolver<NamedParameters, Default_solver>::type, bool>::value),
+      "Eigen3 version 3.2 or later is required.");
+#endif
+
+    typedef typename Default_solver::Matrix Eigen_matrix;
+    typedef typename Default_solver::Vector Eigen_vector;
+
+    std::size_t n = static_cast<int>(vertices(pmesh).size());
+    Eigen_matrix A(n, n);
+    Eigen_vector bx(n), by(n), bz(n), Xx(n), Xy(n), Xz(n);
+    std::vector<CGAL::Triple<int, int, double> > stiffness;
+
+    internal::Shape_smoother<PolygonMesh, VertexPointMap, VCMap, Default_solver, GeomTraits>
+        smoother(pmesh, vpmap, vcmap);
+
+    smoother.init_smoothing(faces);
+
+    #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+    t.stop();
+    std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+    std::cout << "calculate_stiffness_matrix_elements... ";
+    t.reset(); t.start();
+    #endif
+
+    smoother.calculate_stiffness_matrix_elements(stiffness);
+
+    #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+    t.stop();
+    std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+    #endif
+
+    for(std::size_t iter = 0; iter < nb_iterations; ++iter)
+    {
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      std::cout << "setup_system... ";
+      t.reset(); t.start();
+      #endif
+
+      smoother.setup_system(A, bx, by, bz, stiffness, time);
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      t.stop();
+      std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+      std::cout << "solve_system... ";
+      t.reset(); t.start();
+      #endif
+
+      smoother.solve_system(A, Xx, Xy, Xz, bx, by, bz);
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      t.stop();
+      std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+      std::cout << "update_mesh... ";
+      t.reset(); t.start();
+      #endif
+
+      smoother.update_mesh(Xx, Xy, Xz);
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      t.stop();
+      std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+      #endif
+    }
+  }
+}
+
+template<typename PolygonMesh, typename NamedParameters>
+void smooth_along_curvature_flow(PolygonMesh& pmesh, const double& time,
+                                 const NamedParameters& np)
+{
+  smooth_along_curvature_flow(faces(pmesh), pmesh, time, np);
+}
+
+template<typename PolygonMesh>
+void smooth_along_curvature_flow(PolygonMesh& pmesh, const double& time)
+{
+  smooth_along_curvature_flow(faces(pmesh), pmesh, time,
+                              parameters::all_default());
+}
+
+// API for Polyhedron demo plugin
+template<typename PolygonMesh, typename FaceRange, typename NamedParameters>
+void solve_mcf(const FaceRange& faces, PolygonMesh& mesh, const double& time,
```

Comment:
this function should not be in the namespace `CGAL::Polygon_mesh_processing`
It should at least be moved to an `internal` nested namespace